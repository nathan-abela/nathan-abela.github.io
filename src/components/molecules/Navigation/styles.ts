import styled from 'styled-components';

const StyledNavigation = styled.nav``;

const List = styled.ul`
	display: flex;
	align-items: center;
	flex-direction: column;
	font-size: 32px;
	line-height: 2;
	font-weight: bold;

	// theme-toggle
	.desktop-only {
		display: none;
		padding-left: 16px;
	}

	& > li a {
		padding: 8px 16px;
		color: ${({ theme }) => theme.text};
		text-decoration: none;
		transition: 0.2s;

		&:hover {
			color: ${({ theme }) => theme.headline};
		}
	}

	@media screen and (min-width: 768px) {
		flex-direction: row;
		font-size: 16px;
		font-weight: normal;

		.desktop-only {
			display: block;
		}
	}
`;

export { StyledNavigation, List };
