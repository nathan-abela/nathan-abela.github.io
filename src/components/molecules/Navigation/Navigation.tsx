import { FC } from 'react';
import Link from 'next/link';
import { List, StyledNavigation } from './styles';

import { ThemeToggle } from 'Atoms/ThemeToggle';

interface NavigationProps {
	onClick?: () => void;
}

const Navigation: FC<NavigationProps> = ({ onClick }) => {
	return (
		<StyledNavigation>
			<List>
				<li>
					<Link href="/">
						<a onClick={onClick}>Home</a>
					</Link>
				</li>
				<li>
					<Link href="/about">
						<a onClick={onClick}>About</a>
					</Link>
				</li>
				<li>
					<Link href="/education">
						<a onClick={onClick}>Education</a>
					</Link>
				</li>
				<li>
					<Link href="/work">
						<a onClick={onClick}>Work</a>
					</Link>
				</li>
				<li>
					<Link href="/portfolio">
						<a onClick={onClick}>Portfolio</a>
					</Link>
				</li>
				<li className="desktop-only">
					<ThemeToggle />
				</li>
			</List>
		</StyledNavigation>
	);
};

export { Navigation };
