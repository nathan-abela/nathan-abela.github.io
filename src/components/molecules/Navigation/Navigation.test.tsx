import { render } from '@testing-library/react';

import { Navigation } from './Navigation';

const mockDispatch = jest.fn();

jest.mock('react-redux', () => ({
	useSelector: jest.fn(),
	useDispatch: () => mockDispatch,
}));

describe('Navigation', () => {
	it('renders correctly', () => {
		const { container } = render(
			// eslint-disable-next-line prettier/prettier
			<Navigation />
		);
		expect(container).toMatchSnapshot();
	});
});
