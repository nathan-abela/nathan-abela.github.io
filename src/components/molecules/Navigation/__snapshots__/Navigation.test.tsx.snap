// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navigation renders correctly 1`] = `
<div>
  <nav
    class="styles__StyledNavigation-sc-1f4kyre-0"
  >
    <ul
      class="styles__List-sc-1f4kyre-1 ivSflr"
    >
      <li>
        <a
          href="/"
        >
          Home
        </a>
      </li>
      <li>
        <a
          href="/about"
        >
          About
        </a>
      </li>
      <li>
        <a
          href="/education"
        >
          Education
        </a>
      </li>
      <li>
        <a
          href="/work"
        >
          Work
        </a>
      </li>
      <li>
        <a
          href="/portfolio"
        >
          Portfolio
        </a>
      </li>
      <li
        class="desktop-only"
      >
        <button
          class="styles__StyledThemeToggle-sc-1b7mai8-0 kHwexB"
          title="Switch to light theme"
        >
          <test-file-stub />
        </button>
      </li>
    </ul>
  </nav>
</div>
`;
